src/main/java/org/opensearch/neuralsearch/processor/DocumentChunkingProcessor.java

```diff
@@ -0,0 +1,244 @@
+        return TYPE;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void validateDocumentChunkingFieldMap(Map<String, Object> fieldMap) {
+        if (fieldMap == null || fieldMap.isEmpty()) {
+            throw new IllegalArgumentException("Unable to create the processor as field_map is null or empty");
+        }
+
+        for (Map.Entry<String, Object> fieldMapEntry : fieldMap.entrySet()) {
+            String inputField = fieldMapEntry.getKey();
+            Object parameters = fieldMapEntry.getValue();
+
+            if (parameters == null) {
+                throw new IllegalArgumentException("parameters for input field [" + inputField + "] is null, cannot process it.");
+            }
+
+            if (!(parameters instanceof Map)) {
+                throw new IllegalArgumentException(
+                    "parameters for input field [" + inputField + "] cannot be cast to [" + String.class.getName() + "]"
+                );
+            }
+
+            Map<String, Object> parameterMap = (Map<String, Object>) parameters;
+
+            // output field must be string
+            if (!(parameterMap.containsKey(OUTPUT_FIELD))) {
+                throw new IllegalArgumentException("parameters for output field [" + OUTPUT_FIELD + "] is null, cannot process it.");
+            }
+
+            Object outputField = parameterMap.get(OUTPUT_FIELD);
+
+            if (!(outputField instanceof String)) {
+                throw new IllegalArgumentException(
+                    "parameters for output field [" + OUTPUT_FIELD + "] cannot be cast to [" + String.class.getName() + "]"
+                );
+            }
+
+            // check non string parameters
+            int chunkingAlgorithmCount = 0;
+            Map<String, Object> chunkerParameters;
+            for (Map.Entry<?, ?> parameterEntry : parameterMap.entrySet()) {
+                if (!(parameterEntry.getKey() instanceof String)) {
+                    throw new IllegalArgumentException("found parameter entry with non-string key");
+                }
+                String parameterKey = (String) parameterEntry.getKey();
+                if (supportedChunkers.contains(parameterKey)) {
+                    chunkingAlgorithmCount += 1;
+                    chunkerParameters = (Map<String, Object>) parameterEntry.getValue();
+                    IFieldChunker chunker = ChunkerFactory.create(parameterKey, analysisRegistry);
+                    chunker.validateParameters(chunkerParameters);
+                }
+            }
+
+            // should only define one algorithm
+            if (chunkingAlgorithmCount == 0) {
+                throw new IllegalArgumentException("chunking algorithm not defined for input field [" + inputField + "]");
+            }
+            if (chunkingAlgorithmCount > 1) {
+                throw new IllegalArgumentException("multiple chunking algorithms defined for input field [" + inputField + "]");
+            }
```

Comment:
Can we change these checks and the `chunkingAlgorithmCount` to simply check the parameterMap.size() == 1?